P5/p5odds.odd

```diff
@@ -496,17 +496,80 @@ $Id$
                                  | tei:moduleSpec/tei:gloss
                                  | tei:moduleSpec/tei:remarks
                                  ">
-                <sch:report test="not( @xml:lang and @versionDate )">
+                <sch:assert test="@xml:lang  and  @versionDate">
                   Error: both the versionDate and xml:lang attributes on "<sch:name/>" are required when it is a child of "<sch:value-of select="local-name(..)"/>".
-                </sch:report>
+                </sch:assert>
+              </sch:rule>
+            </constraint>
+          </constraintSpec>
+          <constraintSpec mode="add" ident="en-first" scheme="schematron">
+            <desc xml:lang="en" versionDate="2022-05-13">Per ticket <ref target="https://github.com/TEIC/TEI/issues/2279">2279</ref>, require
+            that the <val>en</val> translatable element goes first. Note that we would prefer to be
+            able to say <said>a member of att.translatable</said>, but there is currently no way to
+            that, so we enumerate them.</desc>
+            <constraint>
+              <sch:rule context="  tei:elementSpec/tei:desc[1]
+                                 | tei:elementSpec/tei:gloss[1]
+                                 | tei:elementSpec/tei:remarks[1]
+                                 | tei:classSpec/tei:desc[1]
+                                 | tei:classSpec/tei:gloss[1]
+                                 | tei:classSpec/tei:remarks[1]
+                                 | tei:macroSpec/tei:desc[1]
+                                 | tei:macroSpec/tei:gloss[1]
+                                 | tei:macroSpec/tei:remarks[1]
+                                 | tei:attDef/tei:desc[1]
+                                 | tei:attDef/tei:gloss[1]
+                                 | tei:attDef/tei:remarks[1]
+                                 | tei:valItem/tei:desc[1]
+                                 | tei:valItem/tei:gloss[1]
+                                 | tei:valItem/tei:remarks[1]
+                                 | tei:moduleSpec/tei:desc[1]
+                                 | tei:moduleSpec/tei:gloss[1]
+                                 | tei:moduleSpec/tei:remarks[1]
+                                  ">
+                <sch:let name="lang" value="normalize-space( @xml:lang )"/>
+                <sch:assert test="matches( $lang, '^en-?' )">The first <sch:value-of select="name(.)"/> child of a <sch:value-of select="name(..)"/> should have an @xml:lang that indicates it is in English.</sch:assert>
```

Comment:
Since the above rule enforces that there must be an `@xml:lang`, could this be simplified to `lang('en')` (which should capture all variants)?

Reply:
Well, yes, I suppose so. I have for some time now leaned away from using the lang() function because one of my processors (I do not remember which one, nor, more importantly, which version) used to choke on it. (In the sense that it _always_ returned false, not that it crashed.) But last I remember having that problem was years ago (before the pandemic started). So unless someone else pops up and says “yes, avoid lang()”, I think you are right, we should just go for it. (Especially given that we include a copy of Saxon in our Stylesheets repo, it matters for more what _that_ version of Saxon does than any other processor.)

Reply:
As no one did pop up and say “avoid lang()”, I have now implemented @joeytakeda’s suggestion (in cc21f9a52).